tests/integration/linux_bridge_test.py

```diff
@@ -375,16 +376,35 @@ def test_activate_empty_bridge_does_not_blocked_by_dhcp():
         assertlib.assert_state(desired_state)
 
 
+def test_port_vlan_not_implemented(port0_up):
+    bridge_name = TEST_BRIDGE0
+    port_name = port0_up[Interface.KEY][0][Interface.NAME]
+    port_vlan_config = {
+        LinuxBridge.Port.VLAN_SUBTREE: {
+            LinuxBridge.Port.Vlan.TYPE: LinuxBridge.Port.Vlan.ACCESS_TYPE,
+            LinuxBridge.Port.Vlan.TAG: 101,
+        }
+    }
+
+    bridge_state = _create_bridge_subtree_config(
+        (port_name,), port_vlan_config
+    )
+    with pytest.raises(NmstateNotImplementedError):
+        with linux_bridge(bridge_name, bridge_state):
+            pass
+
+
 def _add_port_to_bridge(bridge_state, ifname):
     port_state = yaml.load(BRIDGE_PORT_YAML, Loader=yaml.SafeLoader)
     add_port_to_bridge(bridge_state, ifname, port_state)
 
 
-def _create_bridge_subtree_config(port_names):
+def _create_bridge_subtree_config(port_names, port_extra_state=None):
     bridge_state = yaml.load(BRIDGE_OPTIONS_YAML, Loader=yaml.SafeLoader)
 
     for port in port_names:
         port_state = yaml.load(BRIDGE_PORT_YAML, Loader=yaml.SafeLoader)
+        port_state.update(port_extra_state or {})
```

Comment:
Does not seem to fit here. The input of ports are plural but you add a single “template” for the extras which does not makes sense much IMO.
I suggest until we make some heavier refactoring here, to just add the needed properties after/outside of this function.

Reply:
Fair enough .

I'll be adding it manually on the relevant test.